README.md

```diff
@@ -1,2 +1,41 @@
-# TypescriptTemplate
-An example of how to utilize types for the rg-bot package
+# RG Bot Typescript Template
+
+This template is used to create a very simple bot for the Regression Games: Ultimate Collector challenge using typescript.  
+_Not sure what this is? Visit https://regression.gg for some programming fun!_
+
+## Minimum Requirements for Regression Games
+
+Your bot must have an `index.ts` file with the following code:
+
+```javascript
+import { RGBot } from "rg-bot";
+
+export function configureBot(bot: RGBot) {
+  // Bot logic here
+}
+```
+This defines a `configureBot` function and exposes that function to Regression Games.
+Regression Games uses it as an entrypoint to your bot script, and passes a bot for you to interact with.
+
+Here is an example of the `configureBot` function with some basic logic that will make your bot parrot back
+anything it sees in chat from other players.
+
+```javascript
+export function configureBot(bot: RGBot) {
+
+  // Every time a player says something in the game, 
+  // do something with that player's username and their message
+  bot.on('chat', (username, message) => {
```

Comment:
Should we have types on our anonymous function here?

Reply:
Updated!